admin_guide/registry_data_check_restore.adoc

```diff
@@ -0,0 +1,176 @@
+----
+
+If the {product-title} cluster's database is too big and you do not want make
+full check then you can limit the scan and check only the database or only the storage:
++
+----
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -experimental -restore-mode=check-database
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -experimental -restore-mode=check-storage
+----
+
+You can also limit the namespace you want to scan:
++
+----
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -experimental -restore-mode=check -restore-namespace=myproject
+----
+
+If some image manifest does not exist on the OpenShift Container Registry's
+storage, then you will see a message like follow:
+
+----
+Broken imagestream tag "myproject/ubuntu:latest" in position 0: image "sha256:be159ff0e12a38fd2208022484bee14412680727ec992680b66cdead1ba76d19": unknown blob
+----
+
+or if blob does not exist:
+
+----
+Broken imagestream tag "myproject/ubuntu:latest" in position 0: image blob "sha256:d42beb8ded595df5627ad4ef31bf528a6fdbfbd11d82f9023152738d6b05a7fa": unknown blob
+----
+
+If you have images in which contains nonexistent blob, but which are not
+in any imagestream, then the message will look like:
+
+----
+Broken image "sha256:be159ff0e12a38fd2208022484bee14412680727ec992680b66cdead1ba76d19": unknown blob
+----
+
+If the storage has image that are not in the imagestream, then the
+message will look like this:
+
+----
+Would add image "sha256:be159ff0e12a38fd2208022484bee14412680727ec992680b66cdead1ba76d19" to imagestream "myproject/ubuntu" with tag "lost-found-be159ff0e12a38fd2208022484bee14412680727ec992680b66cdead1ba76d19"
+----
+
+== Restore
+
+Some data can be recovered. Part of the data is not duplicated and
+therefore their recovery is not possible.
+
+=== Restore imagestreams
+
+To recreate lost imagestream tags based on the state of the storage, use the command:
++
+----
+$ oc -n default \
+    exec -i -t "$(oc -n default get pods -l deploymentconfig=docker-registry \
+    -o jsonpath=$'{.items[0].metadata.name}\n')" \
+    -- /usr/bin/dockerregistry -experimental -restore-mode=recover
+----
+
+Unfortunately, imagestream tag names are not stored on the storage.
+Therefore, they can not be restored. Also, if the imagestream tag had a
+history, then it also could not be restored. Tag creation time will also
+be lost. All images from the history will be restored as separate tags.
+The recovered tag names will have the form: `lost-found-<DIGEST>`. 
+
+=== Restore data on storage
+
+The only way to recover data the OpenShift Container Registry's storage
```

Comment:
"The only way to restore data that was lost from the {product-name} Container Registry's storage"